### PR TITLE
Add pyrfr

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by BastianZim** [comment]
@mfeurer This would be another package I'm adding to conda. Let me know if you're interested!

**Comment by BastianZim** [comment]
cc #5656

File: recipes/pyrfr/meta.yaml
Comment (dhirschfeld) [inline recipes/pyrfr/meta.yaml:36]: :+1:

https://github.com/automl/random_forest_run/blob/e51ab097120c25b7a637e48c6cae16351b8db0c5/pyrfr/setup.py#L42-L55

File: recipes/pyrfr/meta.yaml
Comment (dhirschfeld) [inline recipes/pyrfr/meta.yaml:15]: Since CF doesn't build for py2 I don't think this is necessary. I'll let the CF experts weigh in on this though...

File: recipes/pyrfr/meta.yaml
Comment (dhirschfeld) [inline recipes/pyrfr/meta.yaml:28]: :+1:

https://github.com/automl/random_forest_run/blob/e51ab097120c25b7a637e48c6cae16351b8db0c5/pyrfr/setup.py#L42-L55

File: recipes/pyrfr/meta.yaml
Comment (BastianZim) [reply recipes/pyrfr/meta.yaml:15]: No, it's not needed, I've just been adding too many `noarch` packages recently. 😄 I'll remove it.

Comment (BastianZim) [reply recipes/pyrfr/meta.yaml]: I'm just wondering if I should then add this to python itself, although the linter generally advises against it?

Comment (dhirschfeld) [reply recipes/pyrfr/meta.yaml]: IIUC CF builds for `>=py36`. If the package claims to work on any `py>=30` I probably wouldn't bother putting in a python constraint and just let the CF build config handle it.

I can't speak for CF though - I'm not sure of their official policy in this case...

Comment (BastianZim) [reply recipes/pyrfr/meta.yaml]: Yes you're 100% right, I was thinking of the `noarch` approach again. Your approach is correct.

**Comment by BastianZim** [comment]
@dhirschfeld Just to check, do you think it makes sense to switch to GitHub as a source and running the upstream tests? Was just thinking about it but not 100% sure.

File: recipes/pyrfr/meta.yaml
Comment (BastianZim) [inline recipes/pyrfr/meta.yaml:40]: @mfeurer Just to confirm, is this indeed the license of pyrfr? Just because of the file name...

Comment (dhirschfeld) [reply recipes/pyrfr/meta.yaml:40]: That looks to be the license for [`cereal`](https://github.com/USCiLab/cereal/blob/master/LICENSE)

Comment (dhirschfeld) [reply recipes/pyrfr/meta.yaml:40]: That will need to be copied into the recipe folder during the build. If there are more than one I usually copy all 3rd party licenses into a `thirdparty` folder I create in the `$RECIPE_DIR`.

Comment (dhirschfeld) [reply recipes/pyrfr/meta.yaml:40]: The *actual* license might be a bit problematic:
https://github.com/automl/random_forest_run/blob/e51ab097120c25b7a637e48c6cae16351b8db0c5/pyrfr/setup.py#L47

Comment (BastianZim) [reply recipes/pyrfr/meta.yaml:40]: Yeah, it‘s just the one linked by GitHub in the sidebar which is why I used it. But true, the one in setup is what we would need.

Comment (BastianZim) [reply recipes/pyrfr/meta.yaml:40]: Happy to accept any request you might have but I‘d otherwise wait for @mfeurer to chime in since this might need to be specified upstream and I don‘t really like to guess when it comes to licenses. 😄

File: recipes/pyrfr/meta.yaml
Comment (mfeurer) [inline recipes/pyrfr/meta.yaml:22]: Could you please make this more strict and specify that this can't be SWIG 4 or higher; and there's also a comment by a user that there's a minimal SWIG version that has to be installed: https://github.com/automl/random_forest_run/issues/52

File: recipes/pyrfr/meta.yaml
Comment (mfeurer) [reply recipes/pyrfr/meta.yaml:40]: Do we need to create a new release for the re-licensing: https://github.com/automl/random_forest_run/pull/55 ?

File: recipes/pyrfr/meta.yaml
Comment (mfeurer) [inline recipes/pyrfr/meta.yaml:46]: Yes, please add me.

File: recipes/pyrfr/meta.yaml
Comment (BastianZim) [reply recipes/pyrfr/meta.yaml:22]: Has been added, thanks for the heads up.

File: recipes/pyrfr/meta.yaml
Comment (BastianZim) [reply recipes/pyrfr/meta.yaml:40]: I think that would be the best way since conda-forge prefers having the license packed by the upstream maintainers, as required by most licenses (https://conda-forge.org/docs/maintainer/adding_pkgs.html#packaging-the-license-manually).

Comment (BastianZim) [reply recipes/pyrfr/meta.yaml:40]: If it's not possible, let me know as we can also manually inlcude it.

Comment (mfeurer) [reply recipes/pyrfr/meta.yaml:40]: I'm currently figuring out how to do that again

Comment (mfeurer) [reply recipes/pyrfr/meta.yaml:40]: @franchuterivera This PR is about building a conda package for the pyrfr and requires and update to the pyrfr package on pypi. As building a wheel is already on your stack, do you think it would be much overhead automating the building of a source distribution for pypi? It would be great if that was on github actions, too.

Comment (BastianZim) [reply recipes/pyrfr/meta.yaml:40]: Just to add, we can also use a GitHub release for this, as long as it provides a tarball. 
Whether that is hosted in GitHub or PyPI is not too important.

Reference:

- https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos
- https://docs.github.com/en/github/administering-a-repository/managing-releases-in-a-repository

Sample URLs for the simplejson package:
- https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
- https://github.com/simplejson/simplejson/releases/download/{{ version }}/simplejson-{{ version }}.tar.gz
- https://github.com/simplejson/simplejson/archive/v{{ version }}.tar.gz

Comment (mfeurer) [reply recipes/pyrfr/meta.yaml:40]: Thanks for pointing this out. That's probably a bit easier to do than uploading to pypi as no credentials are needed. As the upload is the easy part compared to building the package we'll probably go with that.

**Comment by BastianZim** [comment]
The following licenses are now included:

1. pyrfr
    1.  License type:  BSD-3-Clause
    2. Included manually
    3. https://github.com/automl/random_forest_run/issues/54#issuecomment-777636690
2. Cereal
    1.  License type:  BSD-3-Clause
    2. Included in PyPI
    3. https://github.com/USCiLab/cereal#cereal-has-a-permissive-license
3. rapidxml
    1. License type: Boost (BSL-1.0) and MIT
    2. Included in PyPI
    3. http://rapidxml.sourceforge.net/license.txt
4. rapidjson
    1.  License type:  MIT
    2. Not included
    3. https://github.com/Tencent/rapidjson/#copyright-and-licensing
5. doxy2swig
    1. License type: Unsure
    2. Not included
    3. Found in https://github.com/automl/random_forest_run/blob/master/lib/doxy2swig/LICENSE

Not sure if all of those licenses are needed but I just wanted to list them all to have an overview.

**Comment by BastianZim** [comment]
@mfeurer Just to check, is the new 0.8.1 version the one we should base this package on or is it just to test the release?

**Comment by BastianZim** [comment]
@mfeurer @dhirschfeld If you're both happy I'd declare this ready for review, or is there something that should still be changed?

**Comment by mfeurer** [comment]
I'm happy about this, but can't give any useful feedback as I'm not familiar with conda forge.

**Review by dhirschfeld — APPROVED** [review]
LGTM

**Comment by BastianZim** [comment]
@dhirschfeld Ok no problem, I'll be able to maintain this but just wanted to ask in case you wanted to have push capabilities.

**Comment by BastianZim** [comment]
@conda-forge/help-python-c Would be ready for review, thanks!

@mfeurer Agreed to be maintainer here: https://github.com/conda-forge/staged-recipes/pull/13919#discussion_r574658119
Licenses are explained here: https://github.com/conda-forge/staged-recipes/pull/13919#issuecomment-793032502

File: recipes/pyrfr/meta.yaml
Comment (chrisburr) [inline recipes/pyrfr/meta.yaml:19]: This should probably be in `host` (I assume it's using Boost-Python?)

Comment (BastianZim) [reply recipes/pyrfr/meta.yaml:19]: This comes from here: _Technically, you need Boost if you want to compile the unit tests._ at: https://github.com/automl/random_forest_run#requirements

To be honest, not sure, because it's just for compiling the tests which aren't even included in the PyPI package so it might be that it's not needed at all.

@mfeurer Maybe you can comment on how you're actually using boost? Otherwise, I could just try it in different sections and see where it fails 😄.

Comment (BastianZim) [reply recipes/pyrfr/meta.yaml:19]: @chrisburr From what I can see in the upstream repo boost is mainly used in the tests and by Cereal and not through boost-python: https://github.com/automl/random_forest_run/search?p=1&q=boost

I suppose having it in host would be being on the save side, correct? So it might make sense to move it there just in case.

Comment (BastianZim) [reply recipes/pyrfr/meta.yaml]: Seems to be working, thanks @chrisburr

**Pull request merged**
